Possible improvement in the notebooks.

ncclementi opened the pull request:
This is  all what I have to suggest after doing the tutorial. I finished all the notebooks. I fixed some typos and made changes in some paragraphs. Besides the code changes, already discussed with @tingyu66. I hope it helps, probably I'm missing things, specially in the writing part.

ncclementi committed:
Fixing typos and definitions

* I changed the definition of parent and child in the class Cell
  because it was confusing.

ncclementi committed:
adding missing periods .

ncclementi committed:
Rewriting definitions and fixing typos

Some definitions in the class Cell were confusing, I rewrite them

ncclementi committed:
Clarifying definitions in add_child function

Some arguments had different explanations than before. I re-wrote
them to be consistent.

ncclementi committed:
Some changes in the text

ncclementi committed:
Change some comments in split_cell

ncclementi committed:
Some typos and re-phrasing

ncclementi committed:
Fixing some gramar and definitions

ncclementi committed:
Merge pull request #4 from barbagroup/master

Update my fork of FMM tutorial

ncclementi committed:
Fixing docstrings

Fixing the docstring on get_multipole function.

ncclementi committed:
Rewriting M2M

I wrote again the M2M to make it more compact and easy to read.
The change was in Step 5 and also in the helper.

Please enter the commit message for your changes. Lines starting

ncclementi committed:
Deleting old version of M2M of step 5

ncclementi committed:
rewriting a sentence that was confusing

ncclementi committed:
rewriting a sentence that was confusingwq

ncclementi committed:
Merge branch 'master' of https://github.com/ncclementi/FMM_tutorial

ncclementi committed:
Re-writing a sentence to cut clutter

ncclementi committed:
typo

ncclementi committed:
typo: leafs replaced by leaves

ncclementi committed:
Replacing twig by leaf and small typo.

I replace thw word twig for leaf to be consistent with all the
previous notebooks.

In the description of the criterion for near and far field
it says that r_cell was the length, it should be l_cell.I
changed it.

ncclementi committed:
Modifying a paragraph that was confusing

ncclementi committed:
Adding small explanation and comment

* Explanation: clarifying that the function evaluate is for only
  one target point and eval_potential is the one that calulates
  the whole potential.

* Comment: On evaluate function, distinction between cell.r (radius)
  and r the distance.

ncclementi committed:
Adding docstrings to eval_potential

ncclementi committed:
Fixing some docstrings in treecode_helper.py

ncclementi committed:
Format in docstrings

Adding periods at the end of the sentences and adding spaces when
needed

ncclementi committed:
Cleaning the helper

* Deleting functions we weren't using at all.
* Making consistent the docstrings.
* Changing names to some functions to be consistent with their
  names in the notebooks.

ncclementi committed:
Removing unnecessary functions

ncclementi committed:
Fixing name of a function

ncclementi committed:
fixing a small typo when rename a function

ncclementi committed:
Adding code to run different theta values

Adding bash and python code to run the treecode.py script with
different theta values. We (with Tingyu) decided do it this way
because it's just few lines instead of repeating code.
The bash parts are commented so it's easy to understand.

gforsyth commented:
I'll go through this more closely tomorrow, but just as an initial comment, I think that all references to Python objects should be wrapped in backticks (the ` character) instead of quotation marks.  I feel that it helps a reader distinguish between things that are concepts and things that are part of the code.  

For example in Notebook 4:

> Since each non-empty cell is an instance which has some same properties (eg. cell's center coordinates, its parent, its children, its multipole, particles inside if it's a leaf), we first need to define a class for a cell, and we call this class "Cell". Every cell is an instance of Class "Cell", and all the instances are stored in a list called "cells". Therefore, the root cell is "cells[0]". 

versus

> Since each non-empty cell is an instance which has some same properties (eg. cell's center coordinates, its parent, its children, its multipole, particles inside if it's a leaf), we first need to define a class for a cell, and we call this class `Cell`. Every cell is an instance of Class `Cell`, and all the instances are stored in a list called `cells`. Therefore, the root cell is `cells[0]`.

ncclementi commented:
Yes! I agree I changed some of them, but it seems I skipped some. I'll take care of that!

ncclementi committed:
Wrapping Python objects in backticks (lesson 1)

ncclementi committed:
Comment on multipole terms and link to bitwise operators

* Clarifying we are showing some terms of the multipole
  expansion, but the remaining ones are similar.

* Adding link to Bitwise Operators from python wiki.

ncclementi committed:
Wrapping python objects and functions names in backsticks

Deleting also unnecesary quotations marks and small typos.

Please enter the commit message for your changes. Lines starting

ncclementi committed:
Adding backsticks and making bigger a plot

ncclementi committed:
Adding a comment on running time if you choose bigger problem.
